ipa-core/src/sharding.rs

```diff
@@ -84,6 +86,16 @@ impl<T> IndexMut<ShardIndex> for Vec<T> {
     }
 }
 
+impl LabelValue for ShardIndex {
```

Comment:
Does it makes sense to implement  `LabelValue` for `T: Into<u64>`? Given the restrictions on overlapping impls, that does seem like it could lead to problems down the line. On the other hand, since this isn't a published API, we can always switch back to the set of impls in this PR if the blanket impl becomes a problem.

Reply:
yea, overlapping impls was the reason not to. I ran into this problem in an early implementation and it gets very annoying if you need to implement `LabelValue` a bit differently. Scraping out blanket implementation when you would need that change would probably be less efficient. 

Maybe I can provide a macro to implement `LabelValue` on those types....

Reply:
I'm not opposed to a macro, but I also don't think there's a clear justification for one here vs. just live with a little boilerplate.

Reply:
yea, that is my thinking too - `LabelValue` does not add too much clutter, but provides explicitness